### PR TITLE
feat(@air/zephyr): add in menu state component to give access to the …

### PR DESCRIPTION
**Pull request by andyhqtran**
## Summary
- Add in `DropdownMenuState` component to give us access to the `isExpanded` state provided by `@reach/menu-button`. We need this because our usage mount/unmount from the dom based on hover.

### CODE REVIEW

File: packages/zephyr/package.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
     "@reach/visually-hidden": "^0.11.1",
     "classnames": "^2.2.6",
     "react-contexify": "^5.0.0",
-    "react-select": "^3.1.1"
+    "react-select": "^3.1.1",
+    "react-use": "^15.3.4"
```
Comment (kylemh) [inline packages/zephyr/package.json:73]: Hmmm what should we do about the potential for dep versions splitting from app here...

Should we instead make this a dep of @air/hooks and re-export everything? Should we just delete the dep from app after this version bump? Or just ignore it and hope there's no issue?

File: packages/zephyr/src/Menus/abstractions/DropdownMenu.tsx [hunk 2]
```diff
@@ -7,16 +7,42 @@ import {
...
+ */
+const DropdownMenuStateManager = ({
+  children,
+  isExpanded,
+  onChange,
+}: DropdownMenuStateManagerProps) => {
+  const isPreviouslyExpanded = usePrevious(isExpanded);
+
+  useEffect(() => {
+    if (isExpanded !== isPreviouslyExpanded && typeof isPreviouslyExpanded !== 'undefined') {
```
Comment (kylemh) [inline packages/zephyr/src/Menus/abstractions/DropdownMenu.tsx:38]:
```suggestion
    if (isExpanded !== isPreviouslyExpanded && isUndefined(isPreviouslyExpanded)) {
```

File: packages/zephyr/package.json (hunk 1 above)
Comment (andyhqtran) [reply packages/zephyr/package.json:73]: I would choose to ignore it and hope there's no issue 😆

Comment (kylemh) [reply packages/zephyr/package.json:73]: go for it

**Review by kylemh — APPROVED** [review]

**Pull request merged**